trappy/ftrace.py

```diff
@@ -644,28 +708,33 @@ class FTrace(GenericFTrace):
 
     def __init__(self, path=".", name="", normalize_time=True, scope="all",
                  events=[], window=(0, None), abs_window=(0, None)):
-        super(FTrace, self).__init__(name, normalize_time, scope, events,
-                                     window, abs_window)
+
         self.raw_events = []
         self.trace_path = self.__process_path(path)
+
+        super(FTrace, self).__init__(name, normalize_time, scope, events,
+                                     window, abs_window)
+
+    def _parsing_setup(self):
+        super(FTrace, self)._parsing_setup()
+
+        if self.read_from_dat:
+            self.file_to_parse = self.__generate_trace_txt(self.trace_path)
+
+        # file_to_parse is already set to trace_path in the superclass,
+        # so no "else" is needed here
+
         self.__populate_metadata()
```

Comment:
I think it would be clearer to put `self.__populate_metadata` at the end of `__init__` directly, putting it here gives a false impression that it needs to be done at this specific stage of parsing.

Reply:
SysTrace is also a subclass of GenericFtrace but does not have any `__populate_metadata` method, which is why I put it here.

Reply:
But can't it just go at the end of `FTrace.__init__`

Reply:
Woops, good thing you told me to look at that. Turns out I hadn't updated `__populate_metadata` to use `file_to_parse` instead of `trace_path`, so it wasn't doing anything actually.

Also, yes, you're right, it can be moved there. I thought you were talking about the superclass.

Reply:
Argh ! Metadata aren't being saved in the cache ! This means we were missing out things like trace._cpus, which would explain some of the issues we had with cache. Thanks for pointing me in the right direction, I got some more work to do...